packages/zoe/src/contractFacet/zcfSeat.js

```diff
@@ -113,50 +113,25 @@ export const createSeatManager = (
    * and so can be used internally for reallocations that violate
    * conservation.
    *
-   * @type {ReallocateInternal}
+   * @type {ReallocateForZCFMint}
    */
-  const reallocateInternal = seats => {
-    // This check was already done in `reallocate`, but
-    // zcfMint.mintGains and zcfMint.burnLosses call
-    // `reallocateInternal` directly without calling `reallocate`, so
-    // we must check here as well.
-    seats.forEach(assertActive);
-    seats.forEach(assertStagedAllocation);
-
-    // Keep track of seats used so far in this call, to prevent aliasing.
-    const zcfSeatsSoFar = new Set();
-
-    seats.forEach(seat => {
-      assert(
-        zcfSeatToSeatHandle.has(seat),
-        X`The seat ${seat} was not recognized`,
-      );
-      assert(
-        !zcfSeatsSoFar.has(seat),
-        X`Seat (${seat}) was already an argument to reallocate`,
-      );
-      zcfSeatsSoFar.add(seat);
-    });
-
+  const reallocateForZCFMint = (zcfSeat, newAllocation) => {
     try {
-      // No side effects above. All conditions checked which could have
-      // caused us to reject this reallocation.
       // COMMIT POINT
       // All the effects below must succeed "atomically". Scare quotes because
       // the eventual send at the bottom is part of this "atomicity" even
       // though its effects happen later. The send occurs in the order of
       // updates from zcf to zoe, its effects must occur immediately in zoe
       // on reception, and must not fail.
       //
-      // Commit the staged allocations (currentAllocation is replaced
-      // for each of the seats) and inform Zoe of the
+      // Commit the newAllocation and inform Zoe of the
       // newAllocation.
 
-      seats.forEach(commitStagedAllocation);
+      activeZCFSeats.set(zcfSeat, newAllocation);
 
-      const seatHandleAllocations = seats.map(seat => {
+      const seatHandleAllocations = [zcfSeat].map(seat => {
```

Comment:
Since we're specializing anyway, the special case should avoid the complexity costs that came from generalizing.

```suggestion
      const seatHandleAllocations = [
        {
          seatHandle: zcfSeatToSeatHandle.get(zcfSeat),
          allocation: newAllocation,
        },
      ];
```

Reply:
I like that